helpers/progress.go

```diff
@@ -10,16 +10,30 @@ func (DiscardProgressWriter) Write(p []byte) (int, error) {
 	return len(p), nil
 }
 
-// UpdateTitle doesn't do anything but satisfy implementation
-func (DiscardProgressWriter) UpdateTitle(_ string) {}
+// Close doesn't do anything but satisfy implementation
+func (DiscardProgressWriter) Close() error {
+	return nil
+}
+
+// Updatef doesn't do anything but satisfy implementation
+func (DiscardProgressWriter) Updatef(_ string, _ ...any) {}
+
+// Successf doesn't do anything but satisfy implementation
+func (DiscardProgressWriter) Successf(_ string, _ ...any) {}
+
+// Failf doesn't do anything but satisfy implementation
+func (DiscardProgressWriter) Failf(_ string, _ ...any) {}
 
 // DiscardProgressWriter is a ProgressWriter in which all calls succeed without doing anything
 // Use this or nil or if you don't care about writing progress
 type DiscardProgressWriter struct{}
 
-// ProgressWriter wraps io.Writer, but also includes an updateTitle function to give the user
+// ProgressWriter wraps io.Writer, but also includes functions to give the user
 // additional context on what's going on. Useful in OCI for tracking layers
 type ProgressWriter interface {
-	UpdateTitle(string)
+	Updatef(string, ...any)
+	Successf(string, ...any)
+	Failf(string, ...any)
 	io.Writer
+	io.Closer
```

Comment:
This should follow the pattern favoring small interfaces composed together.

ex.: `io.Writer` and `io.Closer` form `io.WriteCloser`

Instead of adding `io.Writer` to the `ProgressWriter`, there should be a new interface `ProgressWriteCloser`

that composes `ProgressWriter` and `io.Closer`. This is so that implementers can leverage interface upgrades for features.

Reply:
If there is never a case where we _don't_ want `.Close` behavior on `ProgressWriter`, this should be `io.WriteCloser` instead of composing both `io.Writer` and `io.Closer`